source/extensions/filters/http/grpc_json_transcoder/descriptor_pool_builder.cc

```diff
@@ -0,0 +1,156 @@
+#include "source/extensions/filters/http/grpc_json_transcoder/descriptor_pool_builder.h"
+
+using Envoy::Protobuf::FileDescriptorSet;
+using std::string;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcJsonTranscoder {
+
+void DescriptorPoolBuilder::requestDescriptorPool(
+    std::function<void(std::unique_ptr<Protobuf::DescriptorPool>&&)> descriptor_pool_available) {
+  descriptor_pool_available_ = descriptor_pool_available;
+  FileDescriptorSet descriptor_set;
+
+  switch (proto_config_.descriptor_set_case()) {
+  case envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder::
+      DescriptorSetCase::kProtoDescriptor:
+    if (!descriptor_set.ParseFromString(
+            api_->fileSystem().fileReadToEnd(proto_config_.proto_descriptor()))) {
+      throw EnvoyException("transcoding_filter: Unable to parse proto descriptor");
+    }
+    loadFileDescriptorProtos(descriptor_set);
```

Comment:
Do we need to sort dependency in this case.  maybe we can call descriptor_pool_available directly?

Reply:
If the sorting is not needed,  please don't do it.   Even it is on the filter loading time in the main thread, it will delay the envoy to be ready, it will use memory and cpu.

Reply:
Will separate and remove the extra sorting.